compiler/parser/RdrHsSyn.hs

```diff
@@ -3118,6 +3133,106 @@ validateTemplate vtTemplateName tbd@TemplateBodyDecls{..}
            Nothing -> app
            Just (L loc _) -> L loc (unLoc app)
 
+    validateImplementsBlocks ::
+         [Located ParsedImplementsDeclBlock]
+      -> P [ValidImplementsDeclBlock]
+    validateImplementsBlocks = mapM validateImplementsBlock
+
+    validateImplementsBlock :: Located ParsedImplementsDeclBlock -> P ValidImplementsDeclBlock
+    validateImplementsBlock (L _ (ParsedImplementsDeclBlock iface decls)) = do
+      impls <- mapM validateMethodImpl (groupValDecls decls)
+      checkMultipleDecls iface impls
+      mapM_ (checkNumArgs iface) impls
```

Comment:
These two checks aren't strictly necessary, since skipping them would still result in compilation errors further down the pipeline, but I thought it was better to give the errors in terms of what the user actually writes and thus avoid leaking our desugaring